File: backend/internal/model/gateway.go
package model

import (
	"time"
)

// Status represents the health state of a gateway.
type Status string

const (
	StatusOnline   Status = "online"
	StatusOffline  Status = "offline"
	StatusDegraded Status = "degraded"
	StatusUnknown  Status = "unknown"
)

// Gateway represents a registered OpenClaw gateway instance.
type Gateway struct {
	ID          string            `json:"id"`
	Name        string            `json:"name"`
	Description string            `json:"description,omitempty"`
	Endpoint    string            `json:"endpoint"`
	Transport   TransportConfig   `json:"transport"`
	Auth        GatewayAuthConfig `json:"auth"`
	Status      Status            `json:"status"`
	Labels      map[string]string `json:"labels,omitempty"`
	EnrolledAt  time.Time         `json:"enrolled_at"`
	LastSeenAt  *time.Time        `json:"last_seen_at,omitempty"`
	TTLSeconds  *int              `json:"ttl_seconds,omitempty"`
}

// TransportConfig defines how Lobstertank connects to a gateway.
type TransportConfig struct {
	Type   string            `json:"type"` // "https", "tailscale", "headscale", "cloudflare"
	Params map[string]string `json:"params,omitempty"`
}

// GatewayAuthConfig defines how Lobstertank authenticates with a gateway.
type GatewayAuthConfig struct {
	Type      string            `json:"type"` // "token", "mtls", "oidc"
	Params    map[string]string `json:"params,omitempty"`
	SecretRef string            `json:"secret_ref,omitempty"` // URI referencing a secret provider entry
}

// CreateGatewayRequest is the payload for registering a new gateway.
type CreateGatewayRequest struct {
	Name        string            `json:"name"`
	Description string            `json:"description,omitempty"`
	Endpoint    string            `json:"endpoint"`
	Transport   TransportConfig   `json:"transport"`
	Auth        GatewayAuthConfig `json:"auth"`
	Labels      map[string]string `json:"labels,omitempty"`
	TTLSeconds  *int              `json:"ttl_seconds,omitempty"`
}

// UpdateGatewayRequest is the payload for updating an existing gateway.
type UpdateGatewayRequest struct {
	Name        *string            `json:"name,omitempty"`
	Description *string            `json:"description,omitempty"`
	Endpoint    *string            `json:"endpoint,omitempty"`
	Transport   *TransportConfig   `json:"transport,omitempty"`
	Auth        *GatewayAuthConfig `json:"auth,omitempty"`
	Labels      map[string]string  `json:"labels,omitempty"`
	TTLSeconds  *int               `json:"ttl_seconds,omitempty"`
}

// HealthCheckResult is returned when probing a gateway.
type HealthCheckResult struct {
	GatewayID string `json:"gateway_id"`
	Status    Status `json:"status"`
	Latency   string `json:"latency,omitempty"`
	Error     string `json:"error,omitempty"`
	CheckedAt string `json:"checked_at"`
}

File: frontend/src/types/gateway.ts
export type GatewayStatus = "online" | "offline" | "degraded" | "unknown";

export interface TransportConfig {
  type: "https" | "tailscale" | "headscale" | "cloudflare";
  params?: Record<string, string>;
}

export interface GatewayAuthConfig {
  type: "token" | "mtls" | "oidc";
  params?: Record<string, string>;
  secret_ref?: string;
}

export interface Gateway {
  id: string;
  name: string;
  description?: string;
  endpoint: string;
  transport: TransportConfig;
  auth: GatewayAuthConfig;
  status: GatewayStatus;
  labels?: Record<string, string>;
  enrolled_at: string;
  last_seen_at?: string;
  ttl_seconds?: number;
}

export interface CreateGatewayRequest {
  name: string;
  description?: string;
  endpoint: string;
  transport: TransportConfig;
  auth: GatewayAuthConfig;
  labels?: Record<string, string>;
  ttl_seconds?: number;
}

export interface HealthCheckResult {
  gateway_id: string;
  status: GatewayStatus;
  latency?: string;
  error?: string;
  checked_at: string;
}

export interface FanOutRequest {
  gateway_ids: string[];
  prompt: string;
}

export interface GatewayResult {
  gateway_id: string;
  gateway_name: string;
  response?: string;
  error?: string;
}

export interface FanOutResponse {
  results: GatewayResult[];
}
